### PR TITLE
refactor permalink code and add test fixture; fyi @jellezijlstra @n8u…

### PR DESCRIPTION
**Pull request by jhpoelen**
…pham

hey @JelleZijlstra - Please review my reorganized permalink code and associated text case. It appears that the second test case fail . . .

### CODE REVIEW

File: test/permalink.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+var test = require('tape');
+var permalink = require('../js/permalink.js');
+
+// see https://www.npmjs.com/package/tape for example on how to write unit tests
+
+test('sample test', function (t) {
```
Comment (JelleZijlstra) [inline test/permalink.js:6]: Let's remove the sample test

File: test/permalink.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+});
+
+
+test('permalink no hash', function (t) {
+    t.plan(1);
+    var actual = permalink('https://example.org/', { id: 1234, genus: "Donald", specificEpithet: "duckus"} ); 
+    t.equal(actual, "https://example.org/#genus=Donald&species=duckus&id=1234"); 
+});
+
+test('permalink with existing hash', function (t) {
```
Comment (JelleZijlstra) [inline test/permalink.js:18]: Yes this will currently fail. To fix it we should probably use the standard URL API as you suggested.

**Comment by JelleZijlstra** [comment]
Also, there is a GitHub Action in this repo that's supposed to run this test, but I don't see test output on this PR. Not sure what is missing there.

**Comment by jhpoelen** [comment]
@JelleZijlstra it turned out that the workflow was looking at a branch called "main". Our main branch is "master".

**Comment by jhpoelen** [comment]
Looks like the checks passed now (I made the test pass by using a String.split instead of a char at).

image

**Review by JelleZijlstra — APPROVED** [review]
Thanks! Good catch about the branch name.

**Comment by jhpoelen** [comment]
@JelleZijlstra thanks for reviewing. Do you mind if we rename the main branch to master. The name "master" has some negative connotations that can be easily avoided.

**Comment by JelleZijlstra** [comment]
I'm fine with renaming the main branch, but it's not my decision to make as I don't have commit access to this repo.

**Comment by n8upham** [comment]
That sounds good to me to do this rename.
@JelleZijlstra I also just invited you to the MDD Team for being able to push changes more directly.

**Comment by jhpoelen** [comment]
@JelleZijlstra would you like to do the honors of renaming the main branch to "main" ?

**Comment by jhpoelen** [comment]
I am still working on the remaining #49 issue of permalink not getting updated.

**Comment by JelleZijlstra** [comment]
Not right now, maybe in the next few days. I'm concerned that the configuration for deploying the site to GitHub pages may rely on the branch name, so I'd have to investigate how that config works to make sure we don't break the deployment.

**Comment by jhpoelen** [comment]
@JelleZijlstra ok, makes sense! I'll continue with the permalink stuff.

**Pull request merged**